### PR TITLE
add 10x vdj option

### PR DESCRIPTION
**Pull request by aopisco**

### CODE REVIEW

**Comment by jamestwebber** [comment]
Does this work?

**Comment by aopisco** [comment]
@jamestwebber yap, it's been working on the branch, @AshDangerByrne and @mborja2019 have been using it

**Comment by jamestwebber** [comment]
Okay great, was worried there might be an issue with the cellranger image.

File: src/utilities/alignment/run_10x_vdj.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    "mus": "MM10-PLUS",
+    "mm10-plus": "MM10-PLUS",
+    "mm10-1.2.0": "mm10-1.2.0",
+    "mus-premrna": "mm10-1.2.0-premrna",
+    "mm10-1.2.0-premrna": "mm10-1.2.0-premrna",
+    "hg19-mm10-3.0.0": "hg19-mm10-3.0.0",
+    "microcebus": "MicMur3-PLUS",
+    "gencode.vM19": "gencode.vM19",
+    "GRCh38_premrna": "GRCh38_premrna",
+    "zebrafish-plus": "danio_rerio_plus_STAR2.6.1d",
```
Comment (jamestwebber) [inline src/utilities/alignment/run_10x_vdj.py:34]: I would imagine that most of these references don't work for this script, maybe just remove them? We can add more VDJ references later if people want them

File: src/utilities/alignment/run_10x_vdj.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    "GRCh38-VDJ":"refdata-cellranger-vdj-GRCh38-alts-ensembl-4.0.0",
+}
+deprecated = {
+    "homo": "hg38-plus",
+    "mus": "mm10-plus",
+    "mus-premrna": "mm10-1.2.0-premrna",
+}
```
Comment (jamestwebber) [inline src/utilities/alignment/run_10x_vdj.py:41]: If the above is true, could remove this code too, no need for deprecated options in a new script.

File: src/utilities/alignment/run_10x_vdj.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+
+def get_parser():
+    """ Construct and return the ArgumentParser object that parses input command.
+    """
+
+    parser = argparse.ArgumentParser(
+        prog="run_10x_count.py",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="Run alignment jobs using 10x",
```
Comment (jamestwebber) [inline src/utilities/alignment/run_10x_vdj.py:61]: change this to specify that it's a VDJ alignment

File: src/utilities/alignment/run_10x_vdj.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+        vcpus=64, memory=256000, storage=2000, ecr_image="demuxer"
+    )
+
+
+def get_parser():
+    """ Construct and return the ArgumentParser object that parses input command.
+    """
+
+    parser = argparse.ArgumentParser(
+        prog="run_10x_count.py",
```
Comment (jamestwebber) [inline src/utilities/alignment/run_10x_vdj.py:59]:
```suggestion
        prog="run_10x_vdj.py",
```

File: src/utilities/alignment/run_10x_vdj.py [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+    genome_base_dir = args.root_dir / "genome" / "cellranger"
+    genome_base_dir.mkdir(parents=True)
+
+    # check if the input genome and region are valid
+    if args.taxon in reference_genomes:
+        if args.taxon in deprecated:
+            logger.warn(
+                f"The name '{args.taxon}' will be removed in the future,"
+                f" start using '{deprecated[args.taxon]}'"
+            )
```
Comment (jamestwebber) [inline src/utilities/alignment/run_10x_vdj.py:164]: related to above comments, maybe can be removed

File: src/utilities/alignment/run_10x_vdj.py [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+    parser.add_argument(
+        "--region",
+        default="west",
+        choices=("east", "west"),
+        help=(
+            "Region you're running jobs in."
+            " Should match the location of"
+            " the fastq.gz files"
+        ),
+    )
```
Comment (jamestwebber) [inline src/utilities/alignment/run_10x_vdj.py:120]: I would just remove this option, it should always be `west` now

File: src/utilities/alignment/run_10x_vdj.py [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+
+    # optional arguments
+    parser.add_argument("--cell_count", type=int, default=3000)
+
+    parser.add_argument(
+        "--dobby",
+        action="store_true",
+        help="Use if 10x run was demuxed locally (post November 2019)",
+    )
```
Comment (jamestwebber) [inline src/utilities/alignment/run_10x_vdj.py:109]: probably worth switching to default `True` at this point. Maybe rename the option to `legacy` and reverse the behavior

File: src/utilities/alignment/run_10x_vdj.py [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+
+    # Run cellranger
+    os.chdir(result_path)
+    command = [
+        CELLRANGER,
+        "vdj",
+        "--localmem=240",
+        # "--nosecondary",
+        # "--disable-ui",
+        # f"--expect-cells={args.cell_count}",
```
Comment (jamestwebber) [inline src/utilities/alignment/run_10x_vdj.py:229]:
```suggestion
```

**Review by jamestwebber — CHANGES_REQUESTED** [review]

File: src/utilities/alignment/run_10x_vdj.py (hunk 3 above)
Comment (aopisco) [reply src/utilities/alignment/run_10x_vdj.py:43]: good idea

File: src/utilities/alignment/run_10x_vdj.py (hunk 6 above)
Comment (aopisco) [reply src/utilities/alignment/run_10x_vdj.py:102]: I'm keeping it just in case

File: src/utilities/alignment/run_10x_vdj.py (hunk 7 above)
Comment (aopisco) [reply src/utilities/alignment/run_10x_vdj.py:91]: This is not something I understand fully, so I'd rather not make a mess

**Pull request merged**